File: generate_updates.py
import csv

# Read CSV and generate UPDATE statements
with open('cost_centers.csv', 'r', encoding='utf-8') as f:
    reader = csv.DictReader(f)
    
    print("-- Update new_account_number in vehicles table based on cost_centers.csv")
    print("BEGIN;")
    print()
    
    for row in reader:
        company = row['company'].strip()
        cost_code = row['cost_code'].strip()
        
        # Escape single quotes for SQL
        company_escaped = company.replace("'", "''")
        
        print(f"UPDATE public.vehicles SET new_account_number = '{cost_code}' WHERE TRIM(company) = '{company_escaped}';")
    
    print()
    print("COMMIT;")

File: generate_updates.js
const fs = require('fs');

const csv = fs.readFileSync('cost_centers.csv', 'utf-8');
const lines = csv.split('\n').slice(1); // Skip header

console.log('-- Update new_account_number in vehicles table based on cost_centers.csv');
console.log('BEGIN;\n');

lines.forEach(line => {
    if (!line.trim()) return;
    
    const match = line.match(/^"?([^"]*?)"?,(.+)$/);
    if (!match) return;
    
    let company = match[1].trim();
    const costCode = match[2].trim();
    
    // Escape single quotes for SQL
    company = company.replace(/'/g, "''");
    
    console.log(`UPDATE public.vehicles SET new_account_number = '${costCode}' WHERE TRIM(company) = '${company}';`);
});

console.log('\nCOMMIT;');
